Doc/library/asyncio-eventloop.rst

```diff
@@ -1206,7 +1206,13 @@ Executing code in thread or process pools
                   pool, cpu_bound)
               print('custom process pool', result)
 
-      asyncio.run(main())
+      if __name__ == '__main__':
+          asyncio.run(main())
+
+   Note that on Windows and macOS, the entry point guard may be necessary as the
+   example would otherwise cause a :exc:`RuntimeError`. For more details, see the
+   `Safe importing of main module
+   <https://docs.python.org/3/library/multiprocessing.html#the-spawn-and-forkserver-start-methods>`_.
```

Comment:
Yeah; _please_ don't use hardcoded external links to internal docs locations—they are needlessly klunky, less precise, more verbose, break more easily and don't give you a warning when they do. 

Wrapping everything up in a suggestion, with some minor tweaks:

```suggestion
   Note that the entry point guard (``if __name__ == '__main__'``)
   is required for option 3 due to the peculiarities of :mod:`multiprocessing`,
   which is used by :class:`~concurrent.futures.ProcessPoolExecutor.
   See :ref:`Safe importing of main module <multiprocessing-safe-main-import>`.
```

As Guido mentioned, you'll need to add a ref target label (which I name `.. _multiprocessing-safe-main-import:`) right before the deflist item `Safe importing of main module` under the "spawn and forkserver start methods" section of the `multiprocessing` module docs. I.e., it should look like:

```rst
    However, global variables which are just module level constants cause no
    problems.

.. _multiprocessing-safe-main-import:

Safe importing of main module
```

Reply:
Thanks, I was checking on whether there was a way to do it without a ref label, but it makes sense that adding it guards against any header name changes and so on.